openmdao/core/problem.py

```diff
@@ -759,90 +760,167 @@ def _compute_total_derivs(self, of=None, wrt=None, return_format='flat_dict',
                         ikey = old_input_list[icount]
                         totals[okey][ikey] = -approx_jac[output_name, input_name]
 
+            return totals
+
         # Solve for derivs using linear solver.
-        else:
+        voi_lists = defaultdict(list)
+        v2rhs_name = {}
```

Comment:
Could you use a more descriptive variable name or add some more comments here? This part of the algorithm/code is very tricky, so it would help to be more verbose than usual with the comments here